comp/core/workloadmeta/collectors/internal/kubelet/kubelet.go

```diff
@@ -444,24 +430,14 @@ func extractResources(spec *kubelet.ContainerSpec) workloadmeta.ContainerResourc
 	}
 
 	// extract GPU resource info from the possible GPU sources
-	uniqueGPUVendor := make(map[string]bool)
-
-	resourceKeys := make([]kubelet.ResourceName, 0, len(spec.Resources.Requests))
+	uniqueGPUVendor := make(map[string]struct{})
 	for resourceName := range spec.Resources.Requests {
-		resourceKeys = append(resourceKeys, resourceName)
-	}
-
-	for _, gpuResourceName := range kubelet.GetGPUResourceNames() {
-		for _, resourceKey := range resourceKeys {
-			if strings.HasPrefix(string(resourceKey), string(gpuResourceName)) {
-				if gpuReq, found := spec.Resources.Requests[resourceKey]; found {
-					resources.GPURequest = pointer.Ptr(uint64(gpuReq.Value()))
-					uniqueGPUVendor[extractGPUVendor(gpuResourceName)] = true
-					break
-				}
-			}
+		gpuName, found := gpu.ExtractSimpleGPUName(gpu.ResourceGPU(resourceName))
+		if found {
+			uniqueGPUVendor[gpuName] = struct{}{}
 		}
 	}
+
 	gpuVendorList := make([]string, 0, len(uniqueGPUVendor))
```

Comment:
Same comment about removing the variable `uniqueGPUVendor` and the extra loop

Reply:
Also, should we add a similar test as the one in `comp/core/workloadmeta/collectors/util/kubernetes_resource_parsers/pod_test.go`?